### PR TITLE
@l2succes => Loyalty Applicant submission tracking

### PR DESCRIPTION
**Pull request by mzikherman**
~~Just a little thing to start...this basically loads the segment JS and passes in our key thru sharify.~~

~~This does work in the browser JS console...now to try to actually use it in the react code.~~

This adds the segment JS into the inquiries template, and also directly tracks the successful loyalty submission with the required props.

I did a tiny refactor to the state to just store an array of the selected conversations.

### CODE REVIEW

**Comment by mzikherman** [comment]
Just pushed a commit that adds the typings, so from the React code you can now just do `window.analytics.track(...` or whatever.

File: src/apps/loyalty/server/template.tsx [hunk 1]
```diff
@@ -3,13 +3,20 @@ interface TemplateData {
...
   entrypoint: string
   bootstrapData?: string
-  sharify?: string
+  sharify?: any
   baseURL: string
 }
 
 export default ({styles, html, entrypoint, bootstrapData, sharify, baseURL}: TemplateData) => {
   const fontsURL = "//fast.fonts.net/cssapi/f7f47a40-b25b-44ee-9f9c-cfdfc8bb2741.css"
-
```
Comment (l2succes) [inline src/apps/loyalty/server/template.tsx]: we should use `<script type="text/javascript" async="" src="https://cdn.segment.com/analytics.js/v1/FtkEwggd3GeIHHtE5qtHPfkY6upwDNHf/analytics.min.js"></script>` to use the js file already cached by the browser on artsy.net.

Then just call `analytics.load()` in `browser.ts`

Comment (l2succes) [inline src/apps/loyalty/server/template.tsx]: By the way, one main advantage of using the UMD version of `analytics.js` (https://github.com/segmentio/analytics.js) is we can also track events on the server-side.
So for instance

```javascript
import * as analytics from "analytics.js"

export function Inquiries(req: Request, res: Response, next: NextFunction) {
  const { loginPagePath } = artsyPassport.options
  if (!req.user) {
    // Analytics here
    analytics.track("User redirected to login page")
    return res.redirect(req.baseUrl + loginPagePath)
  }

  ...
}
```
Something to consider if we want to track events that occur only on the server-side for instance.

Comment (alloy) [inline src/apps/loyalty/server/template.tsx]: We should look into that once we start integrating @mennenia’s analytics work.

**Comment by mzikherman** [comment]
Ok I think this is ready to go! Unless you want me to do that analytics refactor now @l2succes, I didn't quite follow the other lib recommendation- but I'm down to pair :)

File: src/apps/loyalty/containers/inquiries/index.tsx [hunk 2]
```diff
@@ -103,17 +110,30 @@ export class Inquiries extends React.Component<Props, State> {
...
+      inquiry_id: inquiryId,
+    }
+
+    let selectedConversations: SelectedConversation[]
+
+    if (selected) {
+      selectedConversations = this.state.selected_conversations.concat([selectedConversation])
+    } else {
+      selectedConversations = this.state.selected_conversations.filter(val => val.id !== conversationId)
+    }
```
Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Just an FYI: If you’d use the ternary operator here instead, you wouldn’t need to explicitly type `selectedConversations` again.

File: src/apps/loyalty/containers/inquiries/index.tsx [hunk 3]
```diff
@@ -167,9 +182,21 @@ export class Inquiries extends React.Component<Props, State> {
     alert("Sorry, there was an error with your submission, please try again")
   }
 
+  trackSuccessfulSubmission() {
+    let ids: string[] = []
+    ids = this.state.selected_conversations.map(value => value.id)
```
Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Why init with an empty array first? And I’d expect TS to figure out `ids`’s type with:

```
const ids = this.state.selected_conversations.map(value => value.id)
```

File: src/apps/loyalty/containers/inquiries/index.tsx (hunk 2 above)
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Actually it didn't look like I needed to at all, removing the explicit `SelectedConversation[]` type and it still worked!

Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Yeah, but then isn’t it implicitly typed `any`?

File: src/apps/loyalty/containers/inquiries/index.tsx (hunk 3 above)
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/index.tsx]: I was just trying to be verbose with types! I'll try this version.

File: src/apps/loyalty/containers/inquiries/index.tsx (hunk 2 above)
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/index.tsx]: lol yup!

Yea, ternary keeps the type without the explicit one. You're totally right.

File: src/apps/loyalty/containers/inquiries/index.tsx (hunk 3 above)
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Yup you're totally right, it did!

Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: 📇 🎉 

Besides in function parameters, I personally try to use as few explicit typings as possible and instead rely on the right code paths that will make TS implicitly type properly, like with the above `selectedConversations` variable.

File: src/apps/loyalty/containers/inquiries/index.tsx (hunk 2 above)
Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: 🌮 🍻

File: src/apps/loyalty/containers/inquiries/index.tsx [hunk 4]
```diff
@@ -134,24 +150,19 @@ export class Inquiries extends React.Component<Props, State> {
...
-    const ids: string[] = []
-    const artworks = this.state.selected_artworks
-
-    for (const artworkId of Object.keys(artworks)) {
-      if (artworks[artworkId]) {
-        ids.push(artworkId)
-      }
-    }
+    let conversationIds: string[] = []
+    conversationIds = this.state.selected_conversations.map(value => value.id)
```
Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Oops, hadn’t caught the same pattern being used here before.

**Comment by l2succes** [comment]
@mzikherman Ok dope! We can do the refactor later

**Pull request merged**